DeepSpeech.ipynb

```diff
@@ -1105,8 +1306,123 @@
+    "test_decodings, test_labels = forward(session, ted_lium.test)\n",
+    "_, test_wer = print_wer_report(session, \"Test\", test_decodings, test_labels)"
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Logging Hyper Parameters and Results"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, as training and test are done, we persist the results alongside with the involved hyper parameters for further reporting."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "with open('%s/%s' % (log_dir, 'hyper.json'), 'w') as dump_file:\n",
+    "    json.dump({ \\\n",
+    "        'context': { \\\n",
+    "            'time_started': time_started.isoformat(), \\\n",
+    "            'time_finished': time_finished.isoformat(), \\\n",
+    "            'git_hash': get_git_revision_hash(), \\\n",
+    "            'git_branch': get_git_branch() \\\n",
+    "        }, \\\n",
+    "        'parameters': { \\\n",
+    "            'learning_rate': learning_rate, \\\n",
+    "            'beta1': beta1, \\\n",
+    "            'beta2': beta2, \\\n",
+    "            'epsilon': epsilon, \\\n",
+    "            'training_iters': training_iters, \\\n",
+    "            'batch_size': batch_size, \\\n",
+    "            'validation_step': validation_step, \\\n",
+    "            'dropout_rate': dropout_rate, \\\n",
+    "            'relu_clip': relu_clip, \\\n",
+    "            'n_input': n_input, \\\n",
+    "            'n_context': n_context, \\\n",
+    "            'n_hidden_1': n_hidden_1, \\\n",
+    "            'n_hidden_2': n_hidden_2, \\\n",
+    "            'n_hidden_3': n_hidden_3, \\\n",
+    "            'n_hidden_5': n_hidden_5, \\\n",
+    "            'n_hidden_6': n_hidden_6, \\\n",
+    "            'n_cell_dim': n_cell_dim, \\\n",
+    "            'n_character': n_character, \\\n",
+    "            'num_examples_train': ted_lium.train.num_examples, \\\n",
+    "            'num_examples_validation': ted_lium.validation.num_examples, \\\n",
+    "            'num_examples_test': ted_lium.test.num_examples \\\n",
+    "        }, \\\n",
+    "        'results': { \\\n",
+    "            'duration': duration, \\\n",
+    "            'last_train_wer': last_train_wer, \\\n",
+    "            'last_validation_wer': last_validation_wer, \\\n",
+    "            'test_wer': test_wer \\\n",
+    "        } \\\n",
+    "    }, dump_file, sort_keys=True, indent = 4)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's also re-populate a central JS file, that contains all the dumps at once."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 34,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "written = False\n",
+    "logs_dir = \"logs\"\n",
```

Comment:
There is a ```log``` and a ```logs``` one. I missed to introduce and reuse the ```logs_dir``` already in the upper cell for the helpers.